### ISSUE TITLE
FNet Layer

### ISSUE THREAD

**Issue by RomainBrault** [tensorflow/addons#2509]
**Describe the feature and the current behavior/state.**

I propose to add a new layer, for seq2seq models: FNet. The paper, by google, is rather new but already has a great popularity due to its low memory footprint for large sequences, good performances, and the fact that it is trivial to implement. I can do the implementation and I already contributed to tensorflow_addons (quantiles.pinball).

**Relevant information**
- Are you willing to contribute it (yes/no): yes
*If you wish to contribute, then read the requirements for new contributions in [`CONTRIBUTING.md`](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#requirements-for-new-contributions-to-the-repository)*
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): https://arxiv.org/pdf/2105.03824.pdf
- Does the relavent academic paper exceed 50 citations? (yes/no): no (the paper is rather recent --- 9 of May --- but already has 2 citations.
- Is there already an implementation in another framework? (if so, where):  yes. In TF, Pytorch, Jax, ...
  * Official Google repo (jax): https://github.com/google-research/google-research/tree/master/f_net
  * pytorch: https://github.com/rishikksh20/FNet-pytorch, https://github.com/Selozhd/FNet-tensorflow
  * tensorflow: https://github.com/Selozhd/FNet-tensorflow
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**

layer

**Who will benefit with this feature?**

Any persone interrested in seq2seq networks. (Signal, NLP, ...)

**Any other info.**

Yannic Kilcher's video: https://www.youtube.com/watch?v=JJR3pBl78zw

**Comment by bhack** [comment]
We have a tutorial at https://keras.io/examples/vision/mlp_image_classification/

I think that the citation threshold for this paper Is lower then 50. So we need to wait a little bit.

**Comment by RomainBrault** [comment]
I can start working on the MR as I'm pretty confident it'll reach the number soon :)

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
